packages/@ourworldindata/grapher/src/timeline/TimelineComponent.tsx

```diff
@@ -542,86 +725,225 @@ export class TimelineComponent extends React.Component<TimelineComponentProps> {
     onFocus,
     onBlur,
 }: {
-    type: MarkerType | "hoverMarker"
+    className?: string
     ariaLabel?: string
     offsetPercent: number
     formattedMinTime: string
     formattedMaxTime: string
     formattedCurrTime: string
-    tooltipVisible: boolean
-    tooltipZIndex: number
+    tooltip?: React.ReactNode
+    tabIndex?: number
+    zIndex?: number
     onKeyDown?: React.KeyboardEventHandler<HTMLDivElement>
     onFocus?: React.FocusEventHandler<HTMLDivElement>
     onBlur?: React.FocusEventHandler<HTMLDivElement>
-}): React.ReactElement => {
-    const isInteractive = type !== "hoverMarker"
+}): React.ReactElement {
     return (
         // @ts-expect-error aria-value* fields expect a number, but if we're dealing with daily data,
         // the numeric representation of a date is meaningless, so we pass the formatted date string instead.
         <div
-            className={cx("handle", type)}
-            style={{ left: `${offsetPercent}%`, zIndex: tooltipZIndex }}
+            className={cx("GrapherTimeline__Handle", className)}
+            style={{ left: `${offsetPercent}%`, zIndex }}
             role="slider"
-            tabIndex={isInteractive ? 0 : -1}
+            tabIndex={tabIndex}
             aria-valuemin={castToNumberIfPossible(formattedMinTime)}
             aria-valuenow={castToNumberIfPossible(formattedCurrTime)}
             aria-valuemax={castToNumberIfPossible(formattedMaxTime)}
             aria-valuetext={formattedCurrTime}
             aria-label={ariaLabel}
+            aria-description="Use arrow keys to adjust by one step, Page Up/Down for larger steps, Home/End to jump to minimum/maximum."
             aria-orientation="horizontal"
             onFocus={onFocus}
             onBlur={onBlur}
             onKeyDown={onKeyDown}
         >
-            <Tippy
-                content={formattedCurrTime}
-                theme="grapher-dark"
-                placement="top"
-                visible={tooltipVisible}
-                arrow={true}
-            >
-                <div className="icon" />
-            </Tippy>
+            <div className="GrapherTimeline__HandleKnob" />
+            {tooltip}
+        </div>
+    )
+}
+
+function EditableYearTooltip({
+    type,
+    editHandle,
+    currentTime,
+    formattedTime,
+    onStartEditing,
+    onComplete,
+    onChange,
+    onKeyboardNavigation,
+    onMouseEnter,
+    onMouseLeave,
+    onBlur,
+}: {
+    type: Exclude<MarkerType, MarkerType.Hover>
+    editHandle?: Exclude<MarkerType, MarkerType.Hover>
+    currentTime: number
+    formattedTime: string
+    onStartEditing?: () => void
+    onComplete?: (year?: number) => void
+    onChange?: (year: number) => void
+    onKeyboardNavigation?: (key: string, currentTime: number) => number
+    onMouseEnter?: () => void
+    onMouseLeave?: () => void
+    onBlur?: () => void
+}): React.ReactElement {
+    const [inputValue, setInputValue] = React.useState(currentTime.toString())
```

Comment:
Should we make this a number instead? Given that we're using an `<input type='number'>`, and that's well-supported across browsers, we can just move the `parseIntOrUndefined` step to the actual `setInputValue` call site.

Advantage: This way, we can stop users from entering float numbers, like `2021.1`.
Disadvantage: We might need special handling for `-`, because `-` in and of itself is not a valid integer.

Reply:
It would be much nicer but I can't think of another way to make the '-' show up in the input field

Reply:
Yeah, you're right on that.

One option would be to go with react-aria's [NumberField](https://react-spectrum.adobe.com/react-aria/NumberField.html), which is overall quite nice, but I don't know if it's worth using that when we have custom behavior ourselves for key presses and which values we want to snap to.

One simple option to prevent float numbers from being entered (but not from being pasted, e.g.) would be simply to do: `onKeyDown={(e) => { if (e.key === ".") e.preventDefault() }}`

Reply:
I'll just call e.preventDefault() then, everything else seems overkill to me 👍🏻